webhook.rst

```diff
@@ -99,34 +99,79 @@ file:
-final readonly class WebhookListener implements ConsumerInterface
-{
-    public function consume(RemoteEvent $event): void
+    use Symfony\Component\RemoteEvent\Attribute\AsRemoteEventConsumer;
+    use Symfony\Component\RemoteEvent\Consumer\ConsumerInterface;
+    use Symfony\Component\RemoteEvent\Event\Mailer\MailerDeliveryEvent;
+    use Symfony\Component\RemoteEvent\Event\Mailer\MailerEngagementEvent;
+    use Symfony\Component\RemoteEvent\RemoteEvent;
+
+    #[AsRemoteEventConsumer('mailer_mailgun')]
+    final readonly class WebhookListener implements ConsumerInterface
     {
-        if ($event instanceof MailerDeliveryEvent) {
-            $this->handleMailDelivery($event);
-        } elseif ($event instanceof MailerEngagementEvent) {
-            $this->handleMailEngagement($event);
-        } else {
-            // This is not an email event
-            return;
+        public function consume(RemoteEvent $event): void
+        {
+            if ($event instanceof MailerDeliveryEvent) {
+                $this->handleMailDelivery($event);
+            } elseif ($event instanceof MailerEngagementEvent) {
+                $this->handleMailEngagement($event);
+            } else {
+                // This is not an email event
+                return;
+            }
         }
-    }
 
-    private function handleMailDelivery(MailerDeliveryEvent $event): void
-    {
-        // Handle the mail delivery event
+        private function handleMailDelivery(MailerDeliveryEvent $event): void
+        {
+            // Handle the mail delivery event
+        }
+
+        private function handleMailEngagement(MailerEngagementEvent $event): void
+        {
+            // Handle the mail engagement event
+        }
     }
 
-    private function handleMailEngagement(MailerEngagementEvent $event): void
+Usage in combination with the Notifier component
+------------------------------------------------
+
+The usage of the Webhook component when using a third-party transport in
+the Notifier is very similar to the usage with the Mailer.
+
+Currently, the following third-party sms transports support webhooks:
+
+============ ==========================================
+SMS service  Parser service name
+============ ==========================================
+Twilio       ``notifier.webhook.request_parser.twilio``
+============ ==========================================
+
+.. versionadded:: 6.3
+
+    The support for Twilio was introduced in Symfony 6.3.
+
+For SMS transports, an additional ``SmsEvent`` is available in the RemoteEvent
+consumer::
+
+    use Symfony\Component\RemoteEvent\Attribute\AsRemoteEventConsumer;
+    use Symfony\Component\RemoteEvent\Consumer\ConsumerInterface;
+    use Symfony\Component\RemoteEvent\Event\Sms\SmsEvent;
+    use Symfony\Component\RemoteEvent\RemoteEvent;
+
+    #[AsRemoteEventConsumer('notifier_twilio')]
+    final readonly class WebhookListener implements ConsumerInterface
```

Comment:
Readonly class is for PHP 8.2, as SF 6.3 support 8.1 I think we shouldn't use it in documentation examples

Reply:
Removed final and readonly. Both were me copy/pasting the code from my very strict project standards. 😅 

Good catch!